yellowbrick/features/base.py

```diff
@@ -215,13 +225,58 @@ def fit(self, X, y=None, **kwargs):
         """
         super(DataVisualizer, self).fit(X, y, **kwargs)
 
-        # Store the classes for the legend if they're None.
-        if self.classes_ is None:
-            # TODO: Is this the most efficient method?
-            self.classes_ = [str(label) for label in np.unique(y)]
+        self._determine_target_color_type(y)
+
+        # Compute classes and colors if target type is discrete
+        if self._target_color_type == DISCRETE:
+            # Store the classes for the legend if they're None.
+            if self.classes_ is None:
+                # TODO: Is this the most efficient method?
+                self.classes_ = [str(label) for label in np.unique(y)]
+            
+# This fails in case of PCoords if subsampled target misses some values.
+            # Ensures that classes passed by user is equal to that in target
+#            if len(self.classes_)!=len(np.unique(y)):
+#                raise YellowbrickValueError("Number of unique target is not " 
+#                                            "equal to classes")
+            color_values = resolve_colors(n_colors=len(self.classes_), 
+                                          colormap=self.colormap, colors=self.color)
+            self._colors = dict(zip(self.classes_, color_values))
+            
+        # Compute target range if colors are continuous
+        elif self._target_color_type == CONTINUOUS:
```

Comment:
We should probably also put a:

```python
else:
    raise YellowbrickTypeError("unknown target color type '{}'".format(self._target_color_type))
```

Just for future proofing. Plus I don't really like `if`/`elif` without an `else`, it seems like a code smell to me.

Reply:
Will do it.